src/lib/comms/Makefile

```diff
@@ -1,49 +1,38 @@
-obj-$(COMMS) += comms.mod
+obj-$(NETWORK) += networking.mod
 
-obj-comms-$(COMMS) := \
+obj-networking-$(NETWORK) := \
     sol-comms.o
 
-ifeq (y,$(NETWORK))
-    ifeq (y,$(PLATFORM_RIOTOS))
-    obj-comms-$(COMMS) += \
-        sol-network-riot.o
-    endif
+obj-networking-$(PLATFORM_RIOTOS) += \
+    sol-network-riot.o
 
-    ifeq (y,$(SOL_PLATFORM_LINUX))
-    obj-comms-$(COMMS) += \
-        sol-network-linux.o
-    endif
-endif
+obj-networking-$(PLATFORM_LINUX) += \
+    sol-network-linux.o \
+    sol-socket-linux.o
```

Comment:
please take the time and make these files follow the other implementation pattern "sol-XXXX-impl-linux".